### ISSUE TITLE
how does mala work?

### ISSUE THREAD

**Issue by samdeoxys1** [blackjax-devs/blackjax#221]
I'm having trouble adapting the introduction tutorial to mala. Specifically, applying the kernel one step gives a:
TypeError: unsupported operand type(s) for -: 'dict' and 'dict'

```
import jax
import jax.numpy as jnp
import jax.scipy.stats as stats
import matplotlib.pyplot as plt
import numpy as np
import blackjax

loc, scale = 10, 20
observed = np.random.normal(loc, scale, size=1_000)

def logprob_fn(loc,scale,observed=observed):
    logpdf=stats.norm.logpdf(observed,loc,scale)
    return jnp.sum(logpdf) 

initial_position = {"loc": 1.0, "scale": 2.0}   
logprob = lambda x: logprob_fn(**x)
step_size = 0.00001
mala = blackjax.mala(logprob, step_size)
initial_state = mala.init(initial_position)
mala_kernel = jax.jit(mala.step)

rng_key = jax.random.PRNGKey(1)
new_state=initial_state

rng_key,_ = jax.random.split(rng_key,2)
new_state,info=mala_kernel(rng_key,new_state)
new_state
```

This is solvable by changing the initial_position to an np or jnp array. However, once I do that and change the ** in logprob to *:

```
initial_position = {"loc": 1.0, "scale": 2.0}   
initial_position = np.array([1.0,2.0])
logprob = lambda x: logprob_fn(*x)
step_size = 0.00001
mala = blackjax.mala(logprob, step_size)
initial_state = mala.init(initial_position)
mala_kernel = jax.jit(mala.step)

def inference_loop(rng_key, kernel, initial_state, num_samples):
    @jax.jit
    def one_step(state, rng_key):
        state, _ = kernel(rng_key, state)
        return state, state

    keys = jax.random.split(rng_key, num_samples)
    _, states = jax.lax.scan(one_step, initial_state, keys)

    return states

rng_key = jax.random.PRNGKey(0)
states = inference_loop(rng_key, mala_kernel, initial_state, 10_000)
```

now the inference loop would give an error: 

<img width="965" alt="image" src="https://user-images.githubusercontent.com/64565475/173694911-3ec036f8-4882-4955-92d4-4eac399ee3f6.png">

Thank you!

#### Versions

BlackJAX 0.7.0
Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0]
Jax 0.3.13
Jaxlib 0.3.10

**Comment by rlouf** [comment]
This was fixed in #216 so it should work with dictionaries if you install from the main branch. Will cut a new release soon.

**Comment by rlouf** [comment]
0.8.0 released
